dvc/remote/gdrive/__init__.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import unicode_literals
+
+import os
+import posixpath
+
+from funcy import cached_property
+from ratelimit import limits, sleep_and_retry
+from backoff import on_exception, expo
+
+from dvc.scheme import Schemes
+from dvc.path_info import CloudURLInfo
+from dvc.remote.base import RemoteBASE
+from dvc.config import Config
+from dvc.exceptions import DvcException
+from dvc.remote.gdrive.pydrive import (
+    RequestListFile,
+    RequestListFilePaginated,
+    RequestUploadFile,
+    RequestDownloadFile,
+)
+from dvc.remote.gdrive.utils import FOLDER_MIME_TYPE
+
+
+class GDriveURLInfo(CloudURLInfo):
+    @property
+    def netloc(self):
+        return self.parsed.netloc
+
+
+class RemoteGDrive(RemoteBASE):
+    scheme = Schemes.GDRIVE
+    path_cls = GDriveURLInfo
+    REGEX = r"^gdrive://.*$"
+    REQUIRES = {"pydrive": "pydrive"}
+
+    def __init__(self, repo, config):
+        super(RemoteGDrive, self).__init__(repo, config)
+        self.no_traverse = False
+        self.cached_dirs = {}
```

Comment:
makes sense to use cached_property for these two?

Reply:
yes, but current usage logic of  `self.cached_dirs` and `self.cached_ids` seems doesn't allow retrieving them from cached properties:
- Both dictionaries are filled from single invocation of `self.cache_root_dirs`
- Dictionaries should be filled with values only after `self.root_id` is fetched from Google Disk's API. Current implementation, seems, is the most compact (it checks just empty dictionaries for values in comparison to have `if statements` to check value of `self.root_id`)

Reply:
makes sense ... may be it's even better to remove these empty initializers here? the same way you don't root_id for example